autoload/clap/provider/yanks.vim

```diff
@@ -0,0 +1,70 @@
+" Author: Ratheesh S <ratheeshreddy@gmail.com>
+" Description: List the recently yanked/deleted lines
+" Based on : https://github.com/sgur/ctrlp-extensions.vim
+
+let s:save_cpo = &cpo
+set cpo&vim
+
+let s:yanklist = []
+let s:yanks = {}
+
+function! s:uniq(list) abort
+    let seen = []
+    for i in range(len(a:list))
+        if index(seen, a:list[i]) == -1
+            call add(seen, a:list[i])
+        endif
+    endfor
+    return seen
+endfunction
+
+function! s:remove_duplicated(list, entry) abort
+    let dup = index(a:list, a:entry, 0)
+    if dup > -1
+        call remove(a:list, dup)
+    endif
+endfunction
+
+function! s:yanks_collect() abort
+    let yankstr = getreg('"')
+
+    if !empty(s:yanklist) && yankstr == s:yanklist[0]
+        return
+    endif
+
+    call s:remove_duplicated(s:yanklist, yankstr)
```

Comment:
Could be simpified using `filter`, `:h filter()`.

Reply:
replaced function with `filter()` call as suggested.